src/seedu/addressbook/Main.java

```diff
@@ -1,160 +1,140 @@
 package seedu.addressbook;
 
-import static seedu.addressbook.TextUi.*;
-import static seedu.addressbook.storage.StorageFile.*;
+import seedu.addressbook.model.person.ReadOnlyPerson;
+import seedu.addressbook.storage.StorageFile.*;
 
 import seedu.addressbook.commands.*;
 import seedu.addressbook.model.AddressBook;
 import seedu.addressbook.parser.Parser;
 import seedu.addressbook.storage.StorageFile;
+import seedu.addressbook.ui.TextUi;
 
 import java.io.*;
+import java.util.List;
 
-/* ==============NOTE TO STUDENTS======================================
- * This class header comment below is brief because details of how to
- * use this class are documented elsewhere.
- * ====================================================================
- */
 
 /**
  * Entry point of the address book application.
- * Contains command execution logic; initialises and uses the different components.
+ * Initializes the application and starts the interaction with the user.
  */
 public class Main {
 
     /**
-     * Default file path used if the user doesn't provide the file name.
+     * Version info of the program.
      */
-    public static final String DEFAULT_STORAGE_FILEPATH = "addressbook.txt";
+    public static final String VERSION = "AddessBook Level 2 - Version 1.0";
 
-    /**
-     * These message is not in the TextUi class because program launch and initialisation messages may be shown
-     * outside the given UI (should be logged at launch console), and needs to be shown before the UI is ready.
-     */
-    public static final String MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE = "Launch arguments format: [STORAGE_FILE_PATH]";
-    public static final String MESSAGE_INIT_FAILED = "Failed to initialise address book application. Exiting...";
+    private TextUi ui;
+    private StorageFile storage;
+    private AddressBook addressBook;
 
     /**
-     * Version info of the program.
+     * The list of person shown to the user most recently.
      */
-    public static final String VERSION = "AddessBook Level 1 - Version 1.0";
+    private List<? extends ReadOnlyPerson> lastShownList = null;
```

Comment:
I wonder if this is the right datatype.  Feels like it is too low level. We can encapsulate it into another class at a later stage (not very important)